### ISSUE TITLE
Custom Params Not Working in yaml section 'params'

### ISSUE THREAD

**Issue by bayonasam** [j3ssie/osmedeus#267]
### **The issue:**

When we add custom parameters with the "--params" parameter, they can then be called from any yaml, however, they cannot be used within the "params" section inside the module:

If we execute the following command, we are assigning a custom variable called "Company" which can be used within the yaml:

`osmedeus scan -m /root/osmedeus-base/workflow/test/validate.yaml -t company.com --params 'Company=COMPANY.DOT.COM'`

However, if this custom variable is called within another variable set in the "params" section of the module, the variable is empty, unlike inherited variables such as "Workspace" and "Workspaces", which can be concatenated together.

In the following screenshot, it shows how I call the following variables/params with the "ExecCmdWithOutput" Script.

- Param "Company" established from the --param parameter in the command line. Works well.
- Param "LegacyParams", which is a concat of different legacy variables in the "params" section established by osmedeus. Works well.
- Param "NewParam", which is a new variable within the "params" section calling "Company". Does not work.

![imagen](https://github.com/j3ssie/osmedeus/assets/29063507/960ff460-3d11-4a29-a852-dbdc43ff9e68)

![imagen](https://github.com/j3ssie/osmedeus/assets/29063507/c1c91edd-1391-4de2-9e3f-f9c772fdf791)

In summary, if a new parameter is set from the command line, it can be called from the yaml, but it cannot be used within the "params" section to concatenate values and work with it.

Thanks for helping in advance <3

**Comment by j3ssie** [comment]
Thanks for raising the issue. I've pushed an updated version that addresses the problem you mentioned.
